[Kim Hanjoo] iP

hjoneweek opened the pull request:
### Assistant Ari
Ari helps you with organizing your tasks. You can:

1. **Add** todos, events, and deadline
2. **Mark** tasks as done
3. **Unmark** tasks
4. **Delete** tasks

The base code is [here](https://github.com/nus-cs2103-AY2324S1/ip)

It is 

- Text-based
- Easy to learn and use

- [ ] Check if you have downloaded it! 😉

> "Organize your tasks." - Kim Hanjoo

`public class Main {`
`    public static void main(String[] args) {`
`        System.out.println("Use it")`
`    }`
`}`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
Level 0

Commit:
Level-1

Commit:
Level-2 Completed

Commit:
Level-3

Commit:
Level-4 Completed

Commit:
A-TextUiTesting Completed

Commit:
Level-5 Level-6 Completed

Commit:
Add Save Function

Commit:
Complete Level 7

Commit:
Add LocalDateTime

Commit:
Complete Level-8

Commit:
Add More OOP Elemet

Commit:
Merge branch 'branch-A-MoreOOP'
Added more OOP Element to Ari

Commit:
Create Duke Package

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'
Adds gradle support

Commit:
Import Gradle

Commit:
Add JUnit Tests

Commit:
Add JavaDoc Comments

Commit:
Merge branch 'branch-A-JavaDoc'
Added JavaDoc Comments

raynertjx commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,209 @@
+     * @return a Task object
+     * @throws DukeException if the text is not formatted correctly.
+     */
+    public static Task textToTask(String text) throws DukeException {
```
Perhaps this method could be renamed to `convertTextToTask`?

raynertjx commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,117 @@
+     * @param index task number
+     * @param tasks TaskList object
+     */
+    public void printAfterMark(int index, TaskList tasks) {
```
Perhaps we could use more intuitive method name like `printAfterMarkTask`? You could extend this to the other methods below.

raynertjx commented on src/test/java/duke/StorageTest.java:
```diff
@@ -0,0 +1,68 @@
+
+public class StorageTest {
+    @Test
+    public void textToTask_test1() throws DukeException {
```
Perhaps your method name for your test could be what you're trying to test. It could be something like `testNormalTodo` or `testInvalidTodo`.

raynertjx commented on src/test/java/duke/TaskListTest.java:
```diff
@@ -0,0 +1,40 @@
+
+public class TaskListTest {
+    @Test
+    public void findTaskUsingKeyword_test1() throws FileNotFoundException {
```
Good name for your test method! I think you could use the same idea to name your other tests.

raynertjx commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,95 @@
+            String identifier = command.substring(0, command.indexOf(" "));
+            String task = command.substring(command.indexOf(" ") + 1);
+            String[] arr = task.split("[/]");
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyyMMdd HHmm");
+            DateTimeFormatter formatter2 = DateTimeFormatter.ofPattern("HHmm");
```
I've noticed that you keep using these 2 lines in multiple places in your code. Perhaps we could abstract this out somewhere, like in a utility/helper class?

raynertjx reviewed (commented):
LGTM! Just some nits to fix. You've kept to most of the code quality guidelines, keep it up! 😄

Commit:
Fix minor errors

Commit:
Add GUI

Commit:
Merge branch 'branch-Level-10'
Add GUI

Commit:
Add assert statements

Assert statements were added to Parser and Storage classes
because they check if there are any errors during runtime.

Let's,
* check whether an array with Event description has 3 elements
  and Deadline description has 2 elements
* check if the filePath is empty or not

hjoneweek committed:
Merge pull request #1 from hjoneweek/branch-A-Assertions

Add assert statements

Commit:
Add snooze (postpone/reschedule) functions

Postpone and reschedule functionalities were added because
this will make easier for users to change date and time of tasks.

Let's,
* allow the parser to take in postpone and reschedule commands
* update the file as dates and time change
* make ui respond appropriately to newly added commands

hjoneweek committed:
Merge pull request #2 from hjoneweek/branch-B-Snooze

Add snooze (postpone/reschedule) functions

Commit:
Improve code quality

Improved code quality will improve readability of the code for the readers.

Let's
* Divided the parseAndExecute method into parse method and execute method in
  command classes
* Created a java class for each valid command and have it extend the Command class
* Stored Command class and its children inside commands package

hjoneweek committed:
Merge pull request #3 from hjoneweek/branch-A-CodeQuality

Improve code quality

Commit:
Add bye command into Parser class

Commit:
Merge branch 'Add-ByeCommand'
Moved handling of bye command into the Parser class from Duke class
for better code quality

Commit:
Add more JUnit tests for TaskList, Parser, and Storage class

Commit:
Merge branch 'branch-A-MoreTesting'
Added more JUnit tests to Parser, Storage, and TaskList class

Commit:
Rename files from Duke to Ari

Commit:
Merge branch 'branch-ChangeName'
Changed file names from duke to ari for better consistancy of the chat bot

Commit:
Fix minor bugs

Commit:
Fix minor bugs and coding standard issues

Commit:
Add User Guide and Ui screenshot

Commit:
Fix minor bug in gradle file
